R/module_init_data.R

```diff
@@ -80,41 +80,48 @@ srv_init_data <- function(id, data, modules, filter = teal_slices()) {
     } else if (inherits(data, "teal_data")) {
       reactiveVal(data)
     } else if (test_reactive(data)) {
-      .fallback_on_failure(this = data, that = reactive(req(FALSE)), label = "Reactive data")
+      data
+    } else if (inherits(data, "qenv.error")) {
+      reactive(data)
     }
 
     if (inherits(data, "teal_data_module")) {
       shinyjs::disable(selector = sprintf(".teal-body:has('#%s') .nav li a", session$ns("content")))
     }
 
     observeEvent(data_validated(), {
-      showNotification("Data loaded successfully.", duration = 5)
-      shinyjs::enable(selector = sprintf(".teal-body:has('#%s') .nav li a", session$ns("content")))
-      if (isTRUE(attr(data, "once"))) {
-        # Hiding the data module tab.
-        shinyjs::hide(
-          selector = sprintf(
-            ".teal-body:has('#%s') a[data-value='teal_data_module']",
-            session$ns("content")
+      # showNotification("Data loaded successfully.", duration = 5)
+
+      if (inherits(data_validated(), "teal_data")) {
```

Comment:
Minor comment: Escape instead of wrapping everything in `if`

```suggestion
      req(inherits(data_validated(), "teal_data"))
      # or
      if (!inherits(data_validated(), "teal_data")) return(NULL)
```